spec/rubocop/cop/style/combinable_defined_spec.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::CombinableDefined, :config do
+  it 'does not register an offense for a single `defined?`' do
+    expect_no_offenses(<<~RUBY)
+      defined?(Foo)
+    RUBY
+  end
+
+  context 'joined by `&&`' do
+    it 'does not register an offense for two separate `defined?`s' do
+      expect_no_offenses(<<~RUBY)
+        defined?(Foo) && defined?(Bar)
+      RUBY
+    end
+
+    it 'does not register an offense for two identical `defined?`s' do
+      # handled by Lint/BinaryOperatorWithIdenticalOperands
+      expect_no_offenses(<<~RUBY)
+        defined?(Foo) && defined?(Foo)
+      RUBY
+    end
+
+    it 'does not register an offense for the same constant with different `cbase`s' do
+      expect_no_offenses(<<~RUBY)
+        defined?(Foo) && defined?(::Foo)
+      RUBY
+    end
+
+    it 'registers an offense for two `defined?`s with same nesting' do
+      expect_offense(<<~RUBY)
+        defined?(Foo) && defined?(Foo::Bar)
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Combine nested `defined?` calls.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        defined?(Foo::Bar)
+      RUBY
+    end
+
+    it 'registers an offense for two `defined?`s with the same nesting and `cbase`' do
+      expect_offense(<<~RUBY)
+        defined?(::Foo) && defined?(::Foo::Bar)
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Combine nested `defined?` calls.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        defined?(::Foo::Bar)
+      RUBY
+    end
+
+    it 'registers an offense for three `defined?`s with same nesting' do
+      expect_offense(<<~RUBY)
+        defined?(Foo) && defined?(Foo::Bar) && defined?(Foo::Bar::Baz)
```

Comment:
Might be good to also cover `defined?(Foo::Bar) && defined?(Foo::Bar::Baz)` with a spec.

Reply:
also `defined?(Foo) && defined?(Foo::Bar) && defined?(Foo::Baz)`